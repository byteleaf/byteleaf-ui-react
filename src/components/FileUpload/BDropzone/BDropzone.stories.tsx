import React from 'react';
import { BDropzone } from './BDropzone';

export default {
  title: 'FileUpload|BDropzone',
};

export const BDropzoneStory = () => (
  <BDropzone
    onChange={() => {}}
    dropzoneText="Drop JSON file here!"
    filesLimit={1}
    acceptedFiles={['application/json']}
    errorText={null}
    maxFileSize={1000}
  />
);

BDropzoneStory.story = {
  name: 'Default',
  parameters: {
    notes: 'A dropzone for adding files',
  },
};

export const BDropzoneMultipleStory = () => (
  <BDropzone
    onChange={() => {}}
    dropzoneText="Drop up to 10 JSON files here!"
    filesLimit={10}
    acceptedFiles={['application/json']}
    errorText={null}
    maxFileSize={1000}
  />
);

BDropzoneMultipleStory.story = {
  name: 'Multiple files',
  parameters: {
    notes: 'A dropzone for adding files',
  },
};

export const BDropzoneErrorStory = () => (
  <BDropzone
    onChange={() => {}}
    dropzoneText="Drop JSON file here!"
    filesLimit={1}
    acceptedFiles={['application/json']}
    errorText="Please select a valid file."
    maxFileSize={1000}
  />
);

BDropzoneErrorStory.story = {
  name: 'Error',
  parameters: {
    notes: 'A dropzone for adding files',
  },
};
