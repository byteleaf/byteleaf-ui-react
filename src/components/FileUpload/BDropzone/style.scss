.byteleaf-dropzone {
  .byteleaf-dropzone-paragraph {
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-weight: 400;
    line-height: 1.33;
    letter-spacing: 0em;
  }

  .MuiChip-root {
    margin-bottom: 8px;
  }
}

.dropzone-error p {
  color: #f44336 !important;
}
