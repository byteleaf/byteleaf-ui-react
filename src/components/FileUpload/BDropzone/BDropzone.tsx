import React from 'react';
import { DropzoneArea } from 'material-ui-dropzone';
import { FormHelperText } from '@material-ui/core';
import './style.scss';

interface DropzoneProps {
  onChange: (files: File[]) => void;
  dropzoneText: string;
  filesLimit: number;
  acceptedFiles: string[];
  errorText: string | null;
  maxFileSize: number;
}

export const BDropzone = ({
  onChange,
  dropzoneText,
  filesLimit,
  acceptedFiles,
  errorText,
  maxFileSize,
}: DropzoneProps) => {
  return (
    <div className={`${errorText ? 'dropzone-error' : ''}`}>
      <DropzoneArea
        useChipsForPreview
        dropzoneClass="byteleaf-dropzone"
        dropzoneParagraphClass="byteleaf-dropzone-paragraph"
        acceptedFiles={acceptedFiles}
        dropzoneText={dropzoneText}
        filesLimit={filesLimit}
        maxFileSize={maxFileSize}
        onChange={onChange}
      />

      {errorText ? <FormHelperText>{errorText}</FormHelperText> : null}
    </div>
  );
};
