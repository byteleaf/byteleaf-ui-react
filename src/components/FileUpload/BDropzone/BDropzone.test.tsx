import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'react-app-polyfill/stable';
import { BDropzone } from './BDropzone';

describe('BDropZone', () => {
  it('renders correctly', async () => {
    const { container } = render(
      <BDropzone
        onChange={() => {}}
        dropzoneText="Drop file here!"
        filesLimit={1}
        acceptedFiles={['application/json']}
        errorText={null}
        maxFileSize={1000}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders an error', async () => {
    const { container } = render(
      <BDropzone
        onChange={() => {}}
        dropzoneText="Drop file here!"
        filesLimit={1}
        acceptedFiles={['application/json']}
        errorText="Error!"
        maxFileSize={1000}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
