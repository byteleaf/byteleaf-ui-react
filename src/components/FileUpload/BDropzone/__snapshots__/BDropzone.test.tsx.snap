// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BDropZone renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <div
      aria-disabled="false"
      class="byteleaf-dropzone DropzoneArea-dropZone-2"
      style="position: relative;"
    >
      <div
        class="DropzoneArea-dropzoneTextStyle-5"
      >
        <p
          class="DropzoneArea-dropzoneParagraph-7 byteleaf-dropzone-paragraph"
        >
          Drop file here!
        </p>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root DropzoneArea-uploadIconSize-6"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
          />
        </svg>
      </div>
      <input
        accept="application/json"
        autocomplete="off"
        multiple=""
        style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px; opacity: 0.00001; pointer-events: none;"
        type="file"
      />
    </div>
  </div>
</div>
`;
